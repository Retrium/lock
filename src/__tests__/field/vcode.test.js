import Immutable from 'immutable';

describe('field/vcode', () => {
  let vcode;
  beforeEach(() => {
    jest.resetModules();

    jest.mock('field/index', () => ({
      setField: jest.fn()
    }));

    vcode = require('field/vcode');
  });
  describe('setVcode()', () => {
    it(`trims code`, () => {
      vcode.setVcode(Immutable.fromJS({}), ' 123456 ');
      const { mock } = require('field/index').setField;
      expect(mock.calls.length).toBe(1);
      expect(mock.calls[0]).toMatchSnapshot();
    });
  });
});
