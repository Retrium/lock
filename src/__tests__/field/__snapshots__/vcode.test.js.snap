// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`field/vcode setVcode() trims code 1`] = `
Array [
  Immutable.Map {},
  "vcode",
  "123456",
]
`;
